p2p/node_info.go

```diff
@@ -124,36 +137,40 @@ func (info DefaultNodeInfo) ValidateBasic() error {
 		channels[ch] = struct{}{}
 	}
 
-	// ensure ListenAddr is good
-	_, err := NewNetAddressString(IDAddressString(info.ID(), info.ListenAddr))
-	return err
+	// Validate Moniker.
+	if !cmn.IsASCIIText(info.Moniker) || cmn.ASCIITrim(info.Moniker) == "" {
+		return fmt.Errorf("info.Moniker must be valid non-empty ASCII text without tabs, but got %v", info.Moniker)
+	}
+
+	// Validate Other.
+	other := info.Other
+	txIndex := other.TxIndex
+	switch txIndex {
+	case "", "on", "off":
+	default:
+		return fmt.Errorf("info.Other.TxIndex should be either 'on' or 'off', got '%v'", txIndex)
+	}
+	// XXX: Should we be more strict about address formats?
```

Comment:
Are they always IP4 addresses?

Reply:
No actually, in theory they could be IPv6. We don't actually do anything with this RPCAddress though, just display it over RPC, so it doesn't matter too much here ...